File: app/src/main/java/com/topjohnwu/magisk/view/dialogs/ManagerInstallDialog.java
package com.topjohnwu.magisk.view.dialogs;

import android.app.Activity;
import android.text.TextUtils;

import androidx.annotation.NonNull;

import com.topjohnwu.magisk.Info;
import com.topjohnwu.magisk.R;
import com.topjohnwu.magisk.utils.DownloadApp;
import com.topjohnwu.magisk.utils.Utils;
import com.topjohnwu.magisk.view.MarkDownWindow;

public class ManagerInstallDialog extends CustomAlertDialog {

    public ManagerInstallDialog(@NonNull Activity a) {
        super(a);
        String name = Utils.INSTANCE.fmt("MagiskManager v%s(%d)",
                Info.remoteManagerVersionString, Info.remoteManagerVersionCode);
        setTitle(a.getString(R.string.repo_install_title, a.getString(R.string.app_name)));
        setMessage(a.getString(R.string.repo_install_msg, name));
        setCancelable(true);
        setPositiveButton(R.string.install, (d, i) -> DownloadApp.upgrade(name));
        if (!TextUtils.isEmpty(Info.managerNoteLink)) {
            setNeutralButton(R.string.app_changelog, (d, i) -> MarkDownWindow.show(a, null, Info.managerNoteLink));
        }
    }
}

File: app/src/main/java/com/topjohnwu/magisk/view/dialogs/ManagerInstallDialog.kt
package com.topjohnwu.magisk.view.dialogs

import android.app.Activity
import com.topjohnwu.magisk.Info
import com.topjohnwu.magisk.R
import com.topjohnwu.magisk.model.download.DownloadService
import com.topjohnwu.magisk.model.entity.internal.Configuration
import com.topjohnwu.magisk.model.entity.internal.DownloadSubject
import com.topjohnwu.magisk.view.MarkDownWindow

class ManagerInstallDialog(a: Activity) : CustomAlertDialog(a) {

    init {
        val subject = DownloadSubject.Manager(Configuration.APK.Upgrade)
        setTitle(a.getString(R.string.repo_install_title, a.getString(R.string.app_name)))
        setMessage(a.getString(R.string.repo_install_msg, subject.title))
        setCancelable(true)
        setPositiveButton(R.string.install) { _, _ ->
            DownloadService(a) { this.subject = subject }
        }
        if (Info.remote.app.note.isNotEmpty()) {
            setNeutralButton(R.string.app_changelog) { _, _ ->
                MarkDownWindow.show(a, null, Info.remote.app.note) }
        }
    }
}
